File: scala/src/main/scala/ConcatenationOfArray.scala
// Problem ID: 1929
// URL: https://leetcode.com/problems/concatenation-of-array/

package leetcode

object ConcatenationOfArray {
    def solution(nums: Array[Int]): Array[Int] = {
        val n = nums.length
        val result = new Array[Int](n * 2);
        for (i <- nums.indices) {
          result(i) = nums(i)
          result(i + n) = nums(i)
        }
        result
    }

  def solution2(nums: Array[Int]): Array[Int] = {
      nums ++ nums
  }
}

File: javascript/src/ConcatenationOfArray.js
// Problem ID: 1929
// URL: https://leetcode.com/problems/concatenation-of-array/

const solution = (nums) => {
    return nums.reduce((a, b, i) => {
        a[i] = a[i + nums.length] = b;
        return a;
    }, Array(nums.length * 2));
}

module.exports = {
    solution,
};
